src/routing_table/prefix.rs

```diff
@@ -23,70 +23,51 @@ use std::hash::{Hash, Hasher};
 #[cfg(test)]
 use std::str::FromStr;
 
-/// A prefix with section version.
-#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Deserialize, Serialize)]
-pub struct VersionedPrefix {
-    prefix: Prefix,
-    version: u64,
-}
-
-impl VersionedPrefix {
-    /// Returns the prefix.
-    pub fn prefix(&self) -> &Prefix {
-        &self.prefix
-    }
-
-    /// Returns the version number.
-    pub fn version(&self) -> u64 {
-        self.version
-    }
-}
-
-impl Into<(Prefix, u64)> for VersionedPrefix {
-    fn into(self) -> (Prefix, u64) {
-        (self.prefix, self.version)
-    }
-}
-
 /// A section prefix, i.e. a sequence of bits specifying the part of the network's name space
 /// consisting of all names that start with this sequence.
 #[derive(Clone, Copy, Default, Eq, Deserialize, Serialize)]
 pub struct Prefix {
     bit_count: u16,
     name: XorName,
+    version: u64,
 }
 
 impl Prefix {
     /// Creates a new `Prefix` with the first `bit_count` bits of `name`. Insignificant bits are all
     /// set to 0. If `bit_count` exceeds the `XOR_NAME_BITS`, then it is reduced to this lower
     /// value.
-    pub fn new(bit_count: usize, name: XorName) -> Prefix {
+    pub fn new(bit_count: usize, name: XorName) -> Self {
         Prefix {
             bit_count: cmp::min(bit_count, XOR_NAME_BITS) as u16,
             name: name.set_remaining(bit_count, false),
+            version: 0,
         }
     }
 
-    /// Returns a `VersionedPrefix` with this prefix and the given version number.
-    pub fn with_version(self, version: u64) -> VersionedPrefix {
-        VersionedPrefix {
-            prefix: self,
-            version: version,
-        }
+    /// Returns this `Prefix` with the given version number.
+    pub fn with_version(mut self, version: u64) -> Self {
+        self.version = version;
+        self
+    }
+
+    /// Returns this `Prefix` with the version number incremented by one.
+    pub fn with_incremented_version(mut self) -> Self {
+        self.version += 1;
+        self
     }
 
     /// Returns `self` with an appended bit: `0` if `bit` is `false`, and `1` if `bit` is `true`. If
     /// `self.bit_count` is already at the maximum for this type, then an unmodified copy of `self`
     /// is returned.
-    pub fn pushed(mut self, bit: bool) -> Prefix {
+    pub fn pushed(mut self, bit: bool) -> Self {
```

Comment:
I am not sure, but maybe a version up by one shall be enforced with pushed/popped method call?

Reply:
Yeah, this occurred to me too. Might be a good idea, not sure either.